src/apple/xcode-manager.ts

```diff
@@ -302,103 +331,971 @@ export class XcodeProject {
+   */
+  findSourceBuildPhaseInTarget(
+    target: PBXNativeTarget,
+  ): XcodeProjectObjectWithId<PBXSourcesBuildPhase> | undefined {
+    debug(`Finding source build phase in target: ${target.name}`);
+    if (!target.buildPhases) {
+      debug('No build phases found for target: ' + target.name);
+      return undefined;
+    }
+    for (const phase of target.buildPhases) {
+      const buildPhaseId = phase.value;
+      const buildPhase = this.objects.PBXSourcesBuildPhase?.[buildPhaseId];
+      if (typeof buildPhase !== 'object') {
+        // Ignore comments
+        continue;
+      }
+      debug(
+        `Found source build phase: ${buildPhaseId} for target: ${target.name}`,
+      );
+      return {
+        id: buildPhaseId,
+        obj: buildPhase,
+      };
+    }
+
+    debug(`No source build phase found for target: ${target.name}`);
+    return undefined;
+  }
+
+  // ================================ FILE HELPERS ================================
+
+  /**
+   * Finds all files in the source build phase of a target.
+   *
+   * @param nativeTarget - The target to find the files in
+   * @returns The files in the source build phase of the target, or an empty array if the target is not found or has no source build phase
+   */
+  findFilesInSourceBuildPhase(
+    nativeTarget: XcodeProjectObjectWithId<PBXNativeTarget>,
+  ): string[] {
+    debug(
+      'Finding files in source build phase for target: ' +
+        nativeTarget.obj.name,
+    );
+    const buildPhase = this.findSourceBuildPhaseInTarget(nativeTarget.obj);
+    if (!buildPhase) {
+      debug(
+        `Sources build phase not found for target: ${nativeTarget.obj.name}`,
+      );
+      return [];
+    }
+    const buildPhaseFiles = buildPhase.obj.files;
+    if (!buildPhaseFiles) {
+      debug(
+        `No files found in sources build phase for target: ${nativeTarget.obj.name}`,
+      );
+      return [];
+    }
+    if (!this.objects.PBXBuildFile) {
+      debug('PBXBuildFile is undefined');
+      return [];
+    }
+
+    const result: string[] = [];
+    for (const file of buildPhaseFiles) {
+      debug(`Resolving build phase file: ${file.value}`);
+      // Find the related build file object
+      const buildFileObj = this.objects.PBXBuildFile[
+        file.value
+      ] as PBXBuildFile;
+      if (!buildFileObj || typeof buildFileObj !== 'object') {
+        debug(`Build file object not found for file: ${file.value}`);
+        continue;
+      }
+      debug(`Build file object found for file: ${file.value}`);
+
+      const buildFileRefId = buildFileObj.fileRef;
+      if (!buildFileRefId) {
+        debug(`File reference not found for file: ${file.value}`);
+        continue;
+      }
+      debug(`Build file reference found for file: ${file.value}`);
+
+      // Find the related file reference object
+      const buildFile = this.objects.PBXFileReference?.[buildFileRefId];
```

Comment:
`l`: When the `this.objects.PBXFileReference` is `undefined`, we iterate over the buildPhaseFiles with no reason and print log messages. Instead, we could add a check before the loop and an early return.